src/thunderbots/software/ai/hl/stp/tactic/free_kick_tactic.cpp

```diff
@@ -0,0 +1,96 @@
+#include "free_kick_tactic.h"
+
+#include "ai/hl/stp/action/chip_action.h"
+#include "ai/hl/stp/action/kick_action.h"
+#include "ai/hl/stp/evaluation/calc_best_shot.h"
+#include "ai/hl/stp/evaluation/deflect_off_enemy_target.h"
+#include "ai/hl/stp/evaluation/indirect_chip.h"
+#include "shared/constants.h"
+#include "util/logger/init.h"
+
+namespace
+{
+    /**
+     * Minimum opening seen by robot for a shot on goal.
+     */
+    const Angle MIN_SHOT_ANGLE = Angle::ofDegrees(15);
+
+    /**
+     * Scaling factor for the chip distance (ball - target)
+     */
+    const double CHIP_DISTANCE_SCALING_FACTOR = 0.8;
+
+    /**
+     * Max ball velocity for this tactic to be not done.
+     */
+    const double MAX_BALL_VELOCITY = 1.5;
+}  // namespace
+
+FreeKickTactic::FreeKickTactic(const World& world, bool loop_forever)
+    : world(world), Tactic(loop_forever)
+{
+}
+
+std::string FreeKickTactic::getName() const
+{
+    return "Free Kick Tactic";
+}
+
+void FreeKickTactic::updateParams(const World& updated_world)
+{
+    this->world = updated_world;
+}
+
+
+double FreeKickTactic::calculateRobotCost(const Robot& robot, const World& world)
+{
+    return dist(robot.position(), world.ball().position());
```

Comment:
We should normalize the value to be in the range [0, 1] so it has the same scale as the other cost functions. I usually divide the distance by the total length of the field, and clamp the value

Reply:
fixed